### ISSUE TITLE
Gephi stuck at starting modules in January 2019.

### ISSUE THREAD

**Issue by leolle** [gephi/gephi#2069]
After running gephi a couple of times, it stuck at starting modules from January.

I have tried a dozen Java versions: java7, java1.8.181, java1.8.191.

## Your Environment

* Version used: Gephi 0.9.2
* Java version: jre1.8.181,  
* Operating System: Windows 10 64bit.

log:
-------------------------------------------------------------------------------
>Log Session: Wednesday, January 9, 2019 3:47:46 PM CST
>System Info: 
  Product Version         = Gephi 0.9.2 201709241107
  Operating System        = Windows 10 version 10.0 running on amd64
  Java; VM; Vendor        = 1.8.0_191; Java HotSpot(TM) 64-Bit Server VM 25.191-b12; Oracle Corporation
  Runtime                 = Java(TM) SE Runtime Environment 1.8.0_191-b12
  Java Home               = C:\Program Files\Java\jre1.8.0_191
  System Locale; Encoding = zh_CN (gephi); GBK
  Home Directory          = C:\Users\wuwei
  Current Directory       = D:\software\Gephi-0.9.2
  User Directory          = C:\Users\wuwei\AppData\Roaming\.gephi\0.9.2\dev
  Cache Directory         = C:\Users\wuwei\AppData\Roaming\.gephi\0.9.2\dev\var\cache
  Installation            = D:\software\Gephi-0.9.2\platform
                            D:\software\Gephi-0.9.2\gephi
                            D:\software\Gephi-0.9.2\platform
  Boot & Ext. Classpath   = C:\Program Files\Java\jre1.8.0_191\lib\resources.jar;C:\Program Files\Java\jre1.8.0_191\lib\rt.jar;C:\Program Files\Java\jre1.8.0_191\lib\sunrsasign.jar;C:\Program Files\Java\jre1.8.0_191\lib\jsse.jar;C:\Program Files\Java\jre1.8.0_191\lib\jce.jar;C:\Program Files\Java\jre1.8.0_191\lib\charsets.jar;C:\Program Files\Java\jre1.8.0_191\lib\jfr.jar;C:\Program Files\Java\jre1.8.0_191\classes;C:\Program Files\Java\jre1.8.0_191\lib\ext\access-bridge-64.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\cldrdata.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\dnsns.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\jaccess.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\jfxrt.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\localedata.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\nashorn.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\sunec.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\sunjce_provider.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\sunmscapi.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\sunpkcs11.jar;C:\Program Files\Java\jre1.8.0_191\lib\ext\zipfs.jar
  Application Classpath   = D:\software\Gephi-0.9.2\platform\lib\boot.jar;D:\software\Gephi-0.9.2\platform\lib\org-openide-modules.jar;D:\software\Gephi-0.9.2\platform\lib\org-openide-util-lookup.jar;D:\software\Gephi-0.9.2\platform\lib\org-openide-util-ui.jar;D:\software\Gephi-0.9.2\platform\lib\org-openide-util.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ar_SA.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_bg.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ca.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_cs.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_de.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_es.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_fr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_gl_ES.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_in_ID.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_it.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ja.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ko.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_nl_NL.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_pl.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_pt_BR.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_pt_PT.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ro.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_ru.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_si.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_sq.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_sv.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_tr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_zh_CN.jar;D:\software\Gephi-0.9.2\platform\lib\locale\boot_zh_TW.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ar_SA.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_bg.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ca.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_cs.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_de.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_es.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_fr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_gl_ES.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_in_ID.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_it.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ja.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ko.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_nl_NL.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_pl.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_pt_BR.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_pt_PT.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ro.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_ru.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_si.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_sq.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_sv.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_tr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_zh_CN.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-modules_zh_TW.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_bg.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_ca.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_cs.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_de.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_es.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_fr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_gl_ES.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_in_ID.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_it.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_ja.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_ko.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_nl_NL.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_pl.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_pt_BR.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_ro.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_ru.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_si.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_sv.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_tr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-lookup_zh_CN.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-ui_ja.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-ui_pt_BR.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-ui_ru.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util-ui_zh_CN.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ar_SA.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_bg.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ca.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_cs.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_de.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_es.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_fr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_gl_ES.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_in_ID.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_it.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ja.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ko.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_nl_NL.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_pl.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_pt_BR.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_pt_PT.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ro.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_ru.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_si.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_sq.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_sv.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_tr.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_zh_CN.jar;D:\software\Gephi-0.9.2\platform\lib\locale\org-openide-util_zh_TW.jar
  Startup Classpath       = D:\software\Gephi-0.9.2\platform\core\asm-all-5.0.1.jar;D:\software\Gephi-0.9.2\platform\core\core-base.jar;D:\software\Gephi-0.9.2\platform\core\core.jar;D:\software\Gephi-0.9.2\platform\core\org-netbeans-libs-asm.jar;D:\software\Gephi-0.9.2\platform\core\org-openide-filesystems-compat8.jar;D:\software\Gephi-0.9.2\platform\core\org-openide-filesystems.jar;D:\software\Gephi-0.9.2\platform\core\locale\core-base_ja.jar;D:\software\Gephi-0.9.2\platform\core\locale\core-base_pt_BR.jar;D:\software\Gephi-0.9.2\platform\core\locale\core-base_ru.jar;D:\software\Gephi-0.9.2\platform\core\locale\core-base_zh_CN.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ar_SA.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_bg.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ca.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_cs.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_de.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_es.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_fr.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_gl_ES.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_in_ID.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_it.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ja.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ko.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_nl_NL.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_pl.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_pt_BR.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_pt_PT.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ro.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_ru.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_si.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_sq.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_sv.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_tr.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_zh_CN.jar;D:\software\Gephi-0.9.2\platform\core\locale\core_zh_TW.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-netbeans-libs-asm_ja.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-netbeans-libs-asm_pt_BR.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-netbeans-libs-asm_ru.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-netbeans-libs-asm_zh_CN.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems-compat8_ja.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems-compat8_pt_BR.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems-compat8_ru.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems-compat8_zh_CN.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ar_SA.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_bg.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ca.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_cs.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_de.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_es.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_fr.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_gl_ES.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_in_ID.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_it.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ja.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ko.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_nl_NL.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_pl.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_pt_BR.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_pt_PT.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ro.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_ru.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_si.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_sq.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_sv.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_tr.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_zh_CN.jar;D:\software\Gephi-0.9.2\platform\core\locale\org-openide-filesystems_zh_TW.jar;D:\software\Gephi-0.9.2\gephi\core\locale\core_gephi.jar
-------------------------------------------------------------------------------

**Comment by eduramiba** [comment]
Hi,
Sometimes modules get corrupted. Please try to delete your gephi user directory folder: https://github.com/gephi/gephi/wiki/Troubleshooting
If still not working, uninstall and reinstall Gephi too.

**Comment by leolle** [comment]
Deleting user folder doesn't help, I even tried 0.9.1 version, it's still
the same issue.

**Comment by lalquier** [comment]
Same here... since about January, Gephi takes an extremely long time to start up (an hour at least) on windows 10, with the 64bit edition.
It freezes at 'Starting Modules' (just before 'INFO [org.netbeans.core.startup.NbEvents]: Turning on modules:' in the log files).

I tried:
- changing JVM
- disabling looking up for new plugin versions
- installing/reinstalling
- disabling openGL
- increasing memory size in the default options

Nothing worked so far..

**Comment by eduramiba** [comment]
Are you using a proxy or similar? We have had some problems because Gephi checks if there are updates during initialization, that needs to be improved.

**Comment by lalquier** [comment]
Not using a proxy by default but I am using it on an enterprise network, so it is possible a service is interfering. I will check the proxy settings in gephi in about an hour (next time it starts up) and confirm back here,

**Comment by aliceolivebrower** [comment]
I'm having this issue too and have tried everything listed on these forums to fix it.

**Comment by lalquier** [comment]
Still having this issue 8 months later. Gephi takes well over an hour to start up. When it ends up crashing suddenly later on, that's infuriating. I have come to suspect Oracle changed something in recent versions of Java that is making Gephi choke or time out on start up. I noticed other Java based tools are taking a long time to start up (several minutes) although not as bad as Gephi. I haven't tried yet to install an old version of Java to test that out but I will soon.

**Comment by lalquier** [comment]
Actually, it looks like this is related to this issue - https://github.com/gephi/gephi/issues/1868
I too see a jogamp_exe_tst* process hanging while gephi takes forever to start... however, killing that secondary process just kills gephi for me, so no luck there. I also tried unpacking the dlls as referenced in this thread for Processing, but I had no luck so far making Gephi recognize their location - https://github.com/processing/processing/issues/4783

**Comment by eduramiba** [comment]
@lalquier We will migrate to LWJGL with new [viz-engine](https://github.com/gephi/viz-engine), meanwhile I guess the right combination needs to be used to run Gephi, and that depends on the OS.

But in summary: Java 8 + make sure the antivirus is not messing with Gephi/jogamp

**Comment by lalquier** [comment]
Thanks for the reply. Looking forward to seeing it working.

In the meantime I will try to make it work using native libraries and using -Djogamp.gluegen.UseTempJarCache=false in the launcher 

We have so much software running on corporate laptops for our safety, it is becoming difficult to pinpoint which one is blocking that temp exe file and figure out how to whitelist it.

**Comment by lonihagen** [comment]
I have window 10. I have the same issue. I tried everything people listed. Nothing works. Still frozen at "Starting Modules" I even tried re-installing OS, but no luck. Gephi worked fine until October of 2019 until I've updated the OS. Now, no Gephi for me with this new fancy computer I've got.

**Comment by thegretch** [comment]
Exactly the same problem. Have tried unintall/install Java and Gephi. No luck. Have it on 3 machines. All worked in early October, none work today.

**Comment by iveta00** [comment]
Guys, I see last comment from 2019. Does anybody have any update? I am using Gephi in my work and for half of my team it just stopped working. Most of us having the same computer with the same configuration. IT admin cannot help. Reinstalling Gaphi does not work.

**Comment by cortazar92** [comment]
Found out today that it gets stuck loading modules in my newer laptop hp omen 15 amd 4800h with gtx 1660ti, but on my older amd laptop without gpu it works just fine.

**Comment by mbastian** [comment]
Thanks for the report. This hang is tied to the old engine's native-library startup probe (blocked by outdated/missing graphics drivers, or in some cases antivirus/corporate security intercepting the native-library check). Gephi 0.11 replaced this with a new visualization engine implementation that changes this startup path.

Could you retry on the latest 0.11.x release? If it still hangs, please open a fresh issue with your OS, driver version, and whether antivirus/corporate security software is active. Closing this for now as superseded by the new engine.
